lib/utils/getCustomProps.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const csstree = require('css-tree');
+
+module.exports = (css) => {
+  // Find all root (html, body, :root) nodes
+  const rootNodes = csstree
+    .findAll(
+      csstree.parse(css),
+      (node) =>
+        node.type === 'Rule' &&
+        node.prelude.children.some((sel) =>
+          sel.children.some((child) =>
+            ['html', 'root', 'body'].includes(child.name),
+          ),
+        ),
+    )
+    .map((decl) => csstree.generate(decl))
+    .join('\n');
+  // Find all CSS custom properties declared in root nodes
+  const customProps = csstree.findAll(
+    csstree.parse(rootNodes),
+    (node) => node.type === 'Declaration' && node.property.startsWith('--'),
+  );
+  return `
+[data-herman-preview="colors"] {
+  ${customProps.map((decl) => csstree.generate(decl)).join(';\n  ')};
+}
```

Comment:
At this point I'm still throwing all the root rules together, then just extracting the custom properties and putting it in one new rule. Do we need to be concerned about conflicts between different blocks?

Reply:
If we keep the restriction to three selectors:
- `:root` should override `html`
- `body` should override both `:root` and `html`
- if there are multiple from the same selector, last takes precedence

Reply:
I'm tempted to say we don't try to enforce precedence, but we add an option for a user-supplied stylesheet as well.

Reply:
Note added in https://github.com/oddbird/sassdoc-theme-herman/commit/fb1b633ba7abab36db7a2ded060dd6116e72c866